controls/radpersistenceframework/persistence-framework-allowed-types.md

```diff
@@ -0,0 +1,114 @@
+{{endregion}}
+
+In cases where internal WPF types need to be added to the AllowedTypes collection, you can obtain them at runtime and utilize the AllowTypes method. One such case is with the native WPF `Selector` class that assigns the `SelectedItems` property to a `SelectedItemCollection` type, which is internal. In this specific scenario, invoking the `AllowInputControls` method will resolve this, however, if you need to allow internal types, you can follow the approach from the following example:
+
+#### __[C#] Allowing internal types via the AllowTypes method__
+{{region persistence-framework-allowed-types-4}}
+    var listBox = new ListBox();
+    manager.AllowTypes(listBox.SelectedItems.GetType());
+{{endregion}}
+
+#### __[VB.NET] Allowing internal types via the AllowTypes method__
+{{region persistence-framework-allowed-types-5}}
+    Dim listBox = New ListBox()
+    manager.AllowTypes(listBox.SelectedItems.[GetType]())
+{{endregion}}
+
+## Allowing Types from the Telerik Assemblies 
+
+Each Telerik assembly has an extension method that will load the types in the AllowedTypes collection of a control, on which it is invoked. In the scope of the RadPersistenceFramework, calling these methods on the PersistenceManager will update its AllowedTypes collection.
+
+The following example will show you how to allow the types that are used in the next __Telerik.Windows.Controls.Docking.dll__ and __Telerik.Windows.Controls.Navigation.dll__ inside the PersistenceManager.
+
+#### __[C#] Allowing the types that are used in the Telerik.Windows.Controls.Docking and Telerik.Windows.Controls.Navigation assemblies__
+{{region persistence-framework-allowed-types-6}}
+    PersistenceManager manager = new PersistenceManager()
+        .AllowDockingControls()
+        .AllowNavigationControls();
+{{endregion}}
+
+#### __[VB.NET] Allowing the types that are used in the Telerik.Windows.Controls.Docking and Telerik.Windows.Controls.Navigation assemblies__
+{{region persistence-framework-allowed-types-7}}
+    Dim manager As PersistenceManager = New PersistenceManager()
+        .AllowDockingControls()
+        .AllowNavigationControls()
+{{endregion}}
+
+## TypeRestored Event
+
+In scenarios where the loaded types are required, prior to adding them to the AllowedTypes collection of the PersistenceManager, you can utilize the `TypeRestored` event. This event will be raised when a type is deserialized. The event arguments are of the type of `TypeRestoredEventArgs` and provide the following API:
+
+* `Type`&mdash;Provides the type that is restored.
+* `AssemblyQualifiedName`&mdash;Provides the type's assembly name.
+
+#### __[C#] Utilizing the TypeRestored event__
+{{region persistence-framework-allowed-types-8}}
+    private List<Type> restoredTypes = new List<Type>();
+
+    private void Manager_TypeRestored(object sender, Persistence.Events.TypeRestoredEventArgs e)
+    {
+        if (e.AssemblyQualifiedName.Contains("MyAssembly"))
```

Comment:
It would not be safe to just check, if an assembly is named "MyAssembly", since a malicious assembly can also be named like so. We can add just a comment here like so:

var type = e.Type; // review the type and decide whether to allow the PersistenceFramework to instantiate it by adding it to the AllowedTypes